### PR TITLE
0.31 release notes

### PR DESCRIPTION
**Pull request by trbromley**

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -127,80 +212,158 @@
   ```
   [(#4126)](https://github.com/PennyLaneAI/pennylane/pull/4126)
 
+* The stochastic parameter-shift gradient transform for pulses, `stoch_pulse_grad`, now
+  supports arbitrary Hermitian generating terms in pulse Hamiltonians.
+  [(4132)](https://github.com/PennyLaneAI/pennylane/pull/4132)
+
+<h3>Improvements 🛠</h3>
```
Comment (soranjh) [inline doc/releases/changelog-dev.md:219]: Could you please add the following:

```
* Added non-cubic lattice support for all electron resource estimation.
  [(3956)](https://github.com/PennyLaneAI/pennylane/pull/3956)
```

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -288,7 +451,7 @@
 
 This release contains contributions from (in alphabetical order):
 
-Venkatakrishnan AnushKrishna
+Venkatakrishnan AnushKrishna,
```
Comment (soranjh) [inline doc/releases/changelog-dev.md:454]: Could you please add `Modjtaba Shokrian-Zini` for adding non-cubic resource estimation [PR](https://github.com/PennyLaneAI/pennylane/pull/3956)?

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (trbromley) [reply doc/releases/changelog-dev.md:219]: Thanks @soranjh, done: 6250756bc8633a388c452e303f2fa19920c5988c.

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -4,239 +4,477 @@
...
+  >>> def f2(p, t): return p * t
+  >>> coeffs = [2., f1, f2]
+  >>> observables =  [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
+  >>> qml.dot(coeffs, observables)
+    (2.0*(PauliX(wires=[0])))
+  + (f1(params_0, t)*(PauliY(wires=[0])))
+  + (f2(params_1, t)*(PauliZ(wires=[0])))
+  ```
+
+<h4>Trace distance is now available in qml.qinfo 💥</h4>
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:172]: @isaacdevlugt this is actually one of the UnitaryHack contributions. We could make this line a h5 level, but I think it's better just to remove.

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -4,19 +4,64 @@
...
+  ```pycon
+  >>> qml.FermiC(0)
+  >>> qml.FermiA(3)
+  ```
+
+  These operators can be multiplied by each other to create a fermionic operator that we call a Fermi word.
+
+  ```
+  >>> qml.FermiC(0) * qml.FermiA(0) * qml.FermiC(3) * qml.FermiA(3)
+  <FermiWord = '0+ 0- 3+ 3-'>
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:27]: @soranjh will this be the repr for 0.31?

File: doc/releases/changelog-dev.md [hunk 5]
```diff
@@ -4,19 +4,64 @@
...
+  ```
+
+  Any fermionic operator, be it a single fermionic creation/annihilation operator, a Fermi word, or a Fermi sentence
+  can be mapped back to the qubit basis by using `qml.jordan_wigner`:
+
+  ```pycon
+  >>> qml.jordan_wigner(h)
+  ((-1.75+0j)*(Identity(wires=[0]))) + ((0.6+0j)*(PauliZ(wires=[0]))) + ((1.15+0j)*(PauliZ(wires=[3])))
+  ```
+
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:64]: It'd be great to have a CTA to check out the [demo](https://github.com/PennyLaneAI/qml/pull/813), we can infer the URL from the filename

**Review by trbromley — COMMENTED** [review]
Thanks @isaacdevlugt

File: doc/releases/changelog-dev.md (hunk 4 above)
Comment (soranjh) [reply doc/releases/changelog-dev.md:27]: No, it will be `a⁺(0) a(0) a⁺(3) a(3)`.

Comment (trbromley) [reply doc/releases/changelog-dev.md:27]: @isaacdevlugt, for context there are still a few tweaks being finished for this feature

Comment (isaacdevlugt) [reply doc/releases/changelog-dev.md:27]: All good!

```suggestion
  <FermiWord = 'a⁺(0) a(0) a⁺(3) a(3)'>
```

File: doc/releases/changelog-dev.md [hunk 6]
```diff
@@ -4,19 +4,64 @@
...
+  >>> qml.FermiC(0) * qml.FermiA(0) * qml.FermiC(3) * qml.FermiA(3)
+  <FermiWord = '0+ 0- 3+ 3-'>
+  ```
+
+  Alternatively, Fermi words can be created via `qml.fermi.FermiWord`:
+
+  ```pycon
+  >>> qml.fermi.FermiWord({(0, 3): '+', (0, 3): '-'})
+  <FermiWord = '0+ 0- 3+ 3-'>
+  ```
```
[the remark this answers is not in the document]
Comment (isaacdevlugt) [reply doc/releases/changelog-dev.md]: Hmm... can I ask why? Maybe I'm mentally emphasizing the parallel to how we do Pauli arithmetic — the analogous "word" and "sentence" creation is meant to be done with `PauliWord` and `PauliSentence`, right?

[the remark this answers is not in the document]
Comment (isaacdevlugt) [reply doc/releases/changelog-dev.md]: It's probably a different crowd of folks who'd go for Pauli arithmetic versus fermionic arithmetic, but the fact that we're trying to sway users away from using analogous things is confusing to me.

File: doc/releases/changelog-0.27.0.md [hunk 7]
```diff
@@ -668,7 +668,7 @@
 * `qml.math.unwrap` no longer creates ragged arrays. Lists remain lists.
   [(#3163)](https://github.com/PennyLaneAI/pennylane/pull/3163)
 
-* New `null.qubit` device. The `null.qubit`performs no operations or memory allocations.
+* New `null.qubit` device. The `null.qubit` performs no operations or memory allocations.
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.27.0.md:671]: 🤠

File: doc/releases/changelog-dev.md [hunk 8]
```diff
@@ -4,66 +4,398 @@
...
-  given subspace.
-  [(#2847)](https://github.com/PennyLaneAI/pennylane/pull/2847)
+  - via `qml.string_to_fermi_word`: create a fermionic operator that represents multiple creation and annihilation operators being 
+    multiplied by each other, which we call a Fermi word.
 
-* A function `pauli.pauli_word_prefactor()` is added to extract the prefactor for a given Pauli word.
-  [(#4164)](https://github.com/PennyLaneAI/pennylane/pull/4164)
+    ```pycon
+    >>> qml.string_to_fermi_word('0+ 1- 0+ 1-')
+    <FermiWord = '0+ 1- 0+ 1-'>
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:26]: I like the use of `string_to_fermi_word`! Are you getting `<FermiWord = '0+ 1- 0+ 1-'>` with the output repr though?

File: doc/releases/changelog-dev.md [hunk 9]
```diff
@@ -4,66 +4,398 @@
...
+    >>> string_to_fermi_word('0^ 1 0^ 1')
+    <FermiWord = '0+ 1- 0+ 1-'>
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:28]: Wait.. I didn't realize we support multiple representations :thinking:

File: doc/releases/changelog-dev.md [hunk 10]
```diff
@@ -4,66 +4,398 @@
...
+    >>> word2 = string_to_fermi_word('3+ 3-')
+    >>> sentence = 1.2 * word1 - 0.345 * word2
+    >>> sentence
+    1.2 * a⁺(0) a(0) a⁺(3) a(3)
+    - 0.345 * a⁺(3) a(3)
+    ```
 
-* Added the `FermiSentence` class to represent a linear combination of fermionic operators.
-  [(#4195)](https://github.com/PennyLaneAI/pennylane/pull/4195)
+  - via `qml.FermiC` and `qml.FermiA`: the fermionic creation and annihilation operators, respectively. These operators are
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:42]: Up for discussion, but I'd mention this option first.

**Review by trbromley — COMMENTED** [review]
Thanks @isaacdevlugt!

File: doc/releases/changelog-dev.md (hunk 8 above)
Comment (soranjh) [reply doc/releases/changelog-dev.md:26]: This example belongs to that PR?

File: doc/releases/changelog-dev.md (hunk 9 above)
Comment (soranjh) [reply doc/releases/changelog-dev.md:28]: I am not sure this example should be here.

File: doc/releases/changelog-dev.md (hunk 8 above)
Comment (soranjh) [reply doc/releases/changelog-dev.md:26]: The output should be:

`a⁺(0) a(1) a⁺(0) a(1)`

File: doc/releases/changelog-dev.md (hunk 9 above)
Comment (soranjh) [reply doc/releases/changelog-dev.md:28]: Yes, we support. The output should be:

`a⁺(0) a(1) a⁺(0) a(1)`

File: doc/releases/changelog-0.31.0.md [hunk 11]
```diff
@@ -4,70 +4,406 @@
...
+  [(#4126)](https://github.com/PennyLaneAI/pennylane/pull/4126)
+
+  ```python
+  def U(params):
+      H = jnp.polyval * qml.PauliZ(0) # time dependent Hamiltonian
+      Um = qml.evolve(H)(params, t=10., dense=False)
+      return qml.matrix(Um)
+  params = jnp.array([[0.5]], dtype=complex)
+  jac = jax.jacobian(U, holomorphic=True)(params)
+  ```
```
[the remark this answers is not in the document]
Comment (Qottmann) [reply doc/releases/changelog-0.31.0.md:348]: Yup, sorry about that it should be
```suggestion
  ```python
  def U(params):
      H = jnp.polyval * qml.PauliZ(0) # time dependent Hamiltonian
      Um = qml.evolve(H, dense=False)(params, t=10.)
      return qml.matrix(Um)
  params = jnp.array([[0.5]], dtype=complex)
  jac = jax.jacobian(U, holomorphic=True)(params)
  ```
```
Alternatively could also use `qml.pulse.ParametrizedEvolution(H, params, t=10., dense=False)`

File: doc/releases/changelog-0.31.0.md [hunk 12]
```diff
@@ -4,72 +4,409 @@
...
+  This allows for computing Jacobians of the unitary directly even when `dense=False`.
+  [(#4126)](https://github.com/PennyLaneAI/pennylane/pull/4126)
+
+  ```python
+  def U(params):
+      H = jnp.polyval * qml.PauliZ(0) # time dependent Hamiltonian
+      Um = qml.evolve(H)(params, t=10., dense=False)
+      return qml.matrix(Um)
+  params = jnp.array([[0.5]], dtype=complex)
+  jac = jax.jacobian(U, holomorphic=True)(params)
```
Comment (trbromley) [inline doc/releases/changelog-0.31.0.md:347]:
```suggestion
  def U(params):
      H = jnp.polyval * qml.PauliZ(0) # time dependent Hamiltonian
      Um = qml.evolve(H, dense=False)(params, t=10.)
      return qml.matrix(Um)
  params = jnp.array([[0.5]], dtype=complex)
  jac = jax.jacobian(U, holomorphic=True)(params)
```

File: doc/releases/changelog-dev.md [hunk 13]
```diff
@@ -0,0 +1,798 @@
...
+    
+  Additionally, any fermionic operator, be it a single fermionic creation/annihilation operator, a Fermi word, or a Fermi sentence,
+  can be mapped to the qubit basis by using `qml.jordan_wigner`:
+
+  TODO: update code example
+
+  ```pycon
+  >>> qml.jordan_wigner(h)
+  ((-1.75+0j)*(Identity(wires=[0]))) + ((0.6+0j)*(PauliZ(wires=[0]))) + ((1.15+0j)*(PauliZ(wires=[3])))
+  ```
```
Comment (isaacdevlugt) [inline doc/releases/changelog-dev.md:79]: Just leaving a note here that I still need to redo this one

File: doc/releases/changelog-dev.md [hunk 14]
```diff
@@ -0,0 +1,779 @@
...
+  open-shell system. This duplicates a similar error in `qchem.Molecule` but makes it clear
+  that the `pyscf` backend can be used for open-shell calculations.
+  [(#4058)](https://github.com/PennyLaneAI/pennylane/pull/4058)
+
+<h4>Next-generation device API</h4>
+
+* The new device interface has been integrated with `qml.execute` for autograd, backpropagation, and no differentiation.
+  [(#3903)](https://github.com/PennyLaneAI/pennylane/pull/3903)
+
+* Support for adjoint differentiation has been added to the `DefaultQubit2` device.
```
Comment (isaacdevlugt) [inline doc/releases/changelog-dev.md:471]: Is this the first time we're mentioning default qubit 2? 🤔 Little confused here

File: doc/releases/changelog-dev.md [hunk 15]
```diff
@@ -0,0 +1,774 @@
...
+* The experimental Device interface `qml.devices.experimental.Device` now requires that the `preprocess` method
+  also returns an `ExecutionConfig` object. This allows the device to choose what `"best"` means for various
+  hyperparameters like `gradient_method` and `grad_on_execution`.
+  [(#4007)](https://github.com/PennyLaneAI/pennylane/pull/4007)
+  [(#4102)](https://github.com/PennyLaneAI/pennylane/pull/4102)
+
+* Gradient transforms with Jax no longer support `argnum`. Use `argnums` instead.
+  [(#4076)](https://github.com/PennyLaneAI/pennylane/pull/4076)
+
+* `qml.collections`, `qml.op_sum`, and `qml.utils.sparse_hamiltonian` have been removed.
```
Comment (isaacdevlugt) [inline doc/releases/changelog-dev.md:650]: @trbromley no PR number here 🤔

File: doc/releases/changelog-dev.md [hunk 16]
```diff
@@ -220,21 +220,6 @@
...
-  >>> def f1(p, t): return p[0] * jnp.sin(p[1] * t)
-  >>> def f2(p, t): return p * t
-  >>> coeffs = [2., f1, f2]
-  >>> observables =  [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
-  >>> qml.dot(coeffs, observables)
-    (2.0*(PauliX(wires=[0])))
-  + (f1(params_0, t)*(PauliY(wires=[0])))
-  + (f2(params_1, t)*(PauliZ(wires=[0])))
-  ```
-
```
Comment (trbromley) [inline doc/releases/changelog-dev.md]: I'd like to keep this one top as it was the top-valued bounty

File: doc/releases/changelog-dev.md (hunk 14 above)
Comment (trbromley) [reply doc/releases/changelog-dev.md]: We discussed offline :+1:

File: doc/releases/changelog-0.31.0.md [hunk 17]
```diff
@@ -4,72 +4,411 @@
...
 
-* Added the `TRZ` qutrit rotation operation, which allows applying a Z rotation on a
-  given subspace.
-  [(#2847)](https://github.com/PennyLaneAI/pennylane/pull/2847)
+  - `qml.FermiC` and `qml.FermiA`: the
+    [fermionic creation](https://docs.pennylane.ai/en/stable/code/api/pennylane.FermiC.html) and
+    [annihilation operators](https://docs.pennylane.ai/en/stable/code/api/pennylane.FermiA.html),
+    respectively. These operators are 
+    defined by passing the index of the orbital that the fermionic operator acts on. For instance, 
+    the operators `a^{\dagger}_0` and `a_3` are respectively constructed as
```
Comment (trbromley) [inline doc/releases/changelog-0.31.0.md:28]:
```suggestion
    the operators `a⁺(0)` and `a(3)` are respectively constructed as
```

File: doc/releases/changelog-0.31.0.md [hunk 18]
```diff
@@ -403,24 +727,24 @@
...
   
-* Fixes a bug where the wire ordering of the `wires` argument to `qml.density_matrix`
+* Fixed a bug where the wire ordering of the `wires` argument to `qml.density_matrix`
   was not taken into account.
   [(#4072)](https://github.com/PennyLaneAI/pennylane/pull/4072)
 
-* Removes a patch in `interfaces/autograd.py` that checks for the `strawberryfields.gbs` device.  That device
-  is pinned to PennyLane <= v0.29.0, so that patch is no longer necessary.
+* A patch in `interfaces/autograd.py` that checks for the `strawberryfields.gbs` device has been removed. 
+  That device is pinned to PennyLane <= v0.29.0, so that patch is no longer necessary.
```
Comment (trbromley) [inline doc/releases/changelog-0.31.0.md:747]:
```suggestion
  That device is pinned to PennyLane <= v0.29.0, so that patch is no longer necessary.
  [(#4089)](https://github.com/PennyLaneAI/pennylane/pull/4089)
```

File: doc/releases/changelog-0.31.0.md [hunk 19]
```diff
@@ -4,72 +4,413 @@
...
+    >>> qml.FermiC(0)
+    a⁺(0)
+    >>> qml.FermiA(3)
+    a(3)
+    ```
 
-* Added the `FermiWord` class to represent a fermionic operator such as
-  :math:`a^\dagger_1 a_0 a^\dagger_2 a_3`.
-  [(#4191)](https://github.com/PennyLaneAI/pennylane/pull/4191)
+    These operators can be composed with (`@`), linearly combined with (`+` and `-`), and multiplied by (`*`) other Fermi operators
```
Comment (soranjh) [inline doc/releases/changelog-0.31.0.md:37]: The `@` is not supported for fermionic operators.

File: doc/releases/changelog-0.31.0.md [hunk 20]
```diff
@@ -4,72 +4,413 @@
...
 
-* Added the `TRZ` qutrit rotation operation, which allows applying a Z rotation on a
-  given subspace.
-  [(#2847)](https://github.com/PennyLaneAI/pennylane/pull/2847)
+  - `qml.FermiC` and `qml.FermiA`: the
+    [fermionic creation](https://docs.pennylane.ai/en/stable/code/api/pennylane.FermiC.html) and
+    [annihilation operators](https://docs.pennylane.ai/en/stable/code/api/pennylane.FermiA.html),
+    respectively. These operators are 
+    defined by passing the index of the orbital that the fermionic operator acts on. For instance, 
+    the operators `a⁺(0)` and `a(3)` are respectively constructed as
```
Comment (soranjh) [inline doc/releases/changelog-0.31.0.md:28]: Should not this be in math format, .i.e., :math:`a^\dagger_0 a_3`?

Comment (trbromley) [reply doc/releases/changelog-0.31.0.md:28]: We can't get the math to render in both md and rst :cry:

File: doc/releases/changelog-0.31.0.md (hunk 19 above)
Comment (trbromley) [reply doc/releases/changelog-0.31.0.md:37]:
```suggestion
    These operators can be composed with (`*`) and linearly combined with (`+` and `-`) other Fermi operators
```
@soranjh does that work?

**Review by isaacdevlugt — APPROVED** [review]
👏
